link-grammar/parse/preparation.c

```diff
@@ -58,8 +64,9 @@ static void setup_connectors(Sentence sent)
 		for (Disjunct *d = sent->word[w].d; d != NULL; d = xd)
 		{
 			xd = d->next;
-			if ((set_dist_fields(d->left, w, -1) < 0) ||
-			    (set_dist_fields(d->right, w, 1) >= (int) sent->length))
+			if ((set_dist_fields(d->left, w, -1, 0) < 0) ||
+			    (set_dist_fields(d->right, w, 1, (int)(sent->length-1)) >=
+			     (int) sent->length))
 			{
```

Comment:
This `if` is always false, because `set_dist_fields` cannot return anything less than 0 or greater than sent length.

Reply:
Note that what is clamped to the range [0, sent_length) is `farthest_word`, while the returned value is `i` (actually `nearest_word`), which may get out of this range in the statement:
`i = set_dist_fields(c->next, w, delta, w_clamp) + delta;`

Reply:
oh right, I misread the `delta`.